packages/voting-contract/src/lib.rs

```diff
@@ -136,7 +137,13 @@ where
             vote,
         }),
     })?;
-
+    BALLOTS_BY_VOTER.update(deps.storage, (&info.sender, proposal_id), |bal| match bal {
```

Comment:
Nit: we don't need to Update here (extra read), as the uniqueness of the key is enforced above.

Also, we could combine these two into [one function "save_ballot"](https://github.com/confio/tgrade-contracts/blob/main/contracts/tgrade-trusted-circle/src/state.rs#L621-L629). Maybe with the "AlreadyVoted" check inside the helper. [Something like this](https://github.com/confio/tgrade-contracts/blob/main/contracts/tgrade-trusted-circle/src/contract.rs#L553-L564)

Reply:
Done, create separate helper and replaced `update` with `save` to reduce read.